packages/firestore/src/api/database.ts

```diff
@@ -1347,10 +1353,10 @@ export class DocumentSnapshot<T = firestore.DocumentData>
         return this._converter.fromFirestore(snapshot, options);
       } else {
         const userDataWriter = new UserDataWriter(
-          this._firestore,
+          this._firestore._databaseId,
           this._firestore._areTimestampsInSnapshotsEnabled(),
-          options.serverTimestamps,
-          /* converter= */ undefined
+          options.serverTimestamps || 'none',
```

Comment:
Should this be done inside of `validateSnapshotOptions`?

Reply:
I like the idea behind this and played around with it: https://gist.github.com/schmidt-sebastian/f1e1432128d2e536b26b0007b07f0ddb

Unfortunately, I think we should not do this. We pass `options` to the converter (in line 1353) and it would surprising to the user if we filled in a default value.